pipelines/Jenkinsfile

```diff
@@ -0,0 +1,104 @@
+import static groovy.json.JsonOutput.toJson
+import static groovy.json.JsonOutput.prettyPrint
+
+@NonCPS
+def static parseEvictableHeroArtifacts(String fetchResponse, Integer max) {
+    def dateFmt = "yyyy-MM-dd'T'HH:mm:ss.SSS'+'"
+    def slurper = new groovy.json.JsonSlurper()
+
+    def artifacts = slurper.parseText(fetchResponse).items;
+    def records = artifacts.collect { [
+        repo: it.name,
+        pull: it.version,
+        lastModified: Date.parse(dateFmt, it.assets?.find().lastModified),
+        downloadUrl: it.assets?.find().downloadUrl
+    ] }
+
+    records.sort{ it.lastModified }
+    def recordsCount = records.size()
+    if (recordsCount <= max) {
+        return []
+    }
+    // Note: Groovy subscript operator is INCLUSIVE on both sides, so offset by 1
+    def lastToEvict = recordsCount - max - 1
+    return records[0..lastToEvict]
+}
+
+pipeline {
+    agent any
+    parameters {
+        booleanParam(name: 'doUpload', defaultValue: false, description: 'Flag controlling execution of upload.')
+        booleanParam(name: 'doEvict', defaultValue: true, description: 'Flag controlling execution of eviction.')
+        // Given a hero zip size of ~1.5 GB, a threshold of '20' should keep us at ~30 GB of storage cost
+        string(name: 'maxEntries', defaultValue: '20', description: 'Maximum number of cached artifacts.')
+    }
+    stages {
+        stage('Upload_Hero_Artifact') {
+            steps {
+                // (TODO) Needs to get plugged into the correct stage of PR build
```

Comment:
are you planning to address these TODOs now or in a future PR?

Reply:
This is a "bonus effort" for "if time permits". I probably should have made this a draft PR, in hindsight. 

I'm going to address these now. I don't want to create more work for us down the road.